src/core/dag.rs

```diff
@@ -677,15 +676,16 @@ impl DAG {
           unsafe {
             *sons_parents.as_ptr() = DLL::singleton(ParentCell::Body(lam));
           }
-          let body = go(*body, ctx, sons_parents);
+          let body = go((*body).clone(), ctx, sons_parents);
           unsafe {
             (*lam.as_ptr()).body = body;
           }
           DAG::Single(lam)
         }
 
-        Term::All(_, uses, name, dom, img) => {
+        Term::All(_, uses, name, terms) => {
           // Allocation and updates
+          let (dom, img) = &*terms;
```

Comment:
Remove `&` to avoid unnecessary clones later on, as per other comment on line 740